cookbook/case_studies/feature_engineering/feast_integration/feast_workflow.py

```diff
@@ -0,0 +1,252 @@
+    FeatureView,
+    FileSource,
+    RepoConfig,
+    ValueType,
+    online_response,
+    registry,
+)
+from flytekit.core.node_creation import create_node
+from feast.infra.offline_stores.file import FileOfflineStoreConfig
+from feast.infra.online_stores.sqlite import SqliteOnlineStoreConfig
+from flytekit import reference_task, task, workflow, Workflow
+from flytekit.extras.sqlite3.task import SQLite3Config, SQLite3Task
+from flytekit.types.file import JoblibSerializedFile
+from flytekit.types.file.file import FlyteFile
+from flytekit.types.schema import FlyteSchema
+from sklearn.model_selection import train_test_split
+from sklearn.naive_bayes import GaussianNB
+from flytekit.configuration import aws
+from feature_eng_tasks import mean_median_imputer, univariate_selection
+from feast_dataobjects import FeatureStore, FeatureStoreConfig
+
+
+logger = logging.getLogger(__file__)
+# TODO: find a better way to define these features.
+FEAST_FEATURES = [
+    "horse_colic_stats:rectal temperature",
+    "horse_colic_stats:total protein",
+    "horse_colic_stats:peripheral pulse",
+    "horse_colic_stats:surgical lesion",
+    "horse_colic_stats:abdominal distension",
+    "horse_colic_stats:nasogastric tube",
+    "horse_colic_stats:outcome",
+    "horse_colic_stats:packed cell volume",
+    "horse_colic_stats:nasogastric reflux PH",
+]
+DATABASE_URI = "https://cdn.discordapp.com/attachments/545481172399030272/861575373783040030/horse_colic.db.zip"
+DATA_CLASS = "surgical lesion"
+
+
+sql_task = SQLite3Task(
+    name="sqlite3.horse_colic",
+    query_template="select * from data",
+    output_schema_type=FlyteSchema,
+    task_config=SQLite3Config(
+        uri=DATABASE_URI,
+        compressed=True,
+    ),
+)
+
+
+@task
+def store_offline(feature_store: FeatureStore, dataframe: FlyteSchema):
+    horse_colic_entity = Entity(name="Hospital Number", value_type=ValueType.STRING)
+
+    horse_colic_feature_view = FeatureView(
+        name="horse_colic_stats",
+        entities=["Hospital Number"],
+        features=[
+            Feature(name="rectal temperature", dtype=ValueType.FLOAT),
+            Feature(name="total protein", dtype=ValueType.FLOAT),
+            Feature(name="peripheral pulse", dtype=ValueType.FLOAT),
+            Feature(name="surgical lesion", dtype=ValueType.STRING),
+            Feature(name="abdominal distension", dtype=ValueType.FLOAT),
+            Feature(name="nasogastric tube", dtype=ValueType.STRING),
+            Feature(name="outcome", dtype=ValueType.STRING),
+            Feature(name="packed cell volume", dtype=ValueType.FLOAT),
+            Feature(name="nasogastric reflux PH", dtype=ValueType.FLOAT),
+        ],
+        batch_source=FileSource(
+            path=str(dataframe.remote_path),
+            event_timestamp_column="timestamp",
+        ),
+        ttl=timedelta(days=1),
+    )
+
+    # Ingest the data into feast
+    feature_store.apply([horse_colic_entity, horse_colic_feature_view])
```

Comment:
does this not update the feature store manifest? or are you saying that we are still mutating the same file?

Reply:
yes, it does mutate the registry (the Feast manifest) and we write the [remote path](https://github.com/flyteorg/flytesnacks/blob/4f1222ad83ab450c0af51092daf329df22c8c715/cookbook/case_studies/feature_engineering/feast_integration/feast_workflow.py#L84) of the FlyteSchema.

Reply:
It's the act of localizing specific Feast's `FeatureView` prior to important operations (like loading historical features or materializing data into the online store) that makes this work.